addons/web/static/src/js/fields/abstract_field.js

```diff
@@ -168,9 +168,12 @@ var AbstractField = Widget.extend({
         // the 'widget' attrs if is is given, and if it is a valid key, with a
         // fallback on the field type, ensuring that the value is formatted and
         // displayed according to the choosen widget, if any.
-        this.formatType = this.attrs.widget in field_utils.format ?
-                            this.attrs.widget :
-                            this.field.type;
+        if (!this.formatType) {
```

Comment:
why is this needed?

Reply:
float_toggle widget have the float_factor formatter. So there is no float_toggle formatter. With the code before, the formatter will be the field type one, as the name of the formatter is not the same as the widget in the registry. 
This commit gives priority to the formatter defined on the widget and fallback on a deduction based on formatter/widget name.

Reply:
usually we simply call super in the sub widget, and then override the specific key after (in this case, formatType)